### PR TITLE
For https://github.com/batect/batect/issues/553

### PR DESCRIPTION
**Pull request by binkley**
Add comparison block for Earthly vs Batect.

### CODE REVIEW

**Comment by binkley** [comment]
I'd love a pass over my writing, especially on:
- Style & tone
- What makes sense for Batect

**Comment by binkley** [comment]
Cross-posting from https://github.com/earthly/earthly/issues/2738

I'll update the PR to include the insights from the Earthly folks

File: docs/getting-started/comparison.md [hunk 1]
```diff
@@ -66,6 +66,47 @@ There are a number of differences between Dojo and Batect:
...
+- Both let you compose your build so everything is not in one file
+- Both offer options for parallelization and caching to speed up your builds
+  (see below on differences for caching)
+- Both let you call out to your existing build, testing, or deployment tooling
+- Both have strong documentation
+
+Where Earthly is different from Batect:
+
+- Earthly is an installed program to your system (Batect is self-contained).
+  There are GitHub Actions to add Earthly to your workflow script
```
Comment (charleskorn) [inline docs/getting-started/comparison.md:91]:
```suggestion
  There are GitHub Actions to add Earthly to your workflow script.
```

File: docs/getting-started/comparison.md [hunk 2]
```diff
@@ -66,6 +66,47 @@ There are a number of differences between Dojo and Batect:
...
+- Earthly is both a commercial product, and an open-source project.
+  It provides paid-for plans to use their CI infrastructure, however it also
+  works fine as a standalone program in your local and CI environments
```
Comment (charleskorn) [inline docs/getting-started/comparison.md:94]:
```suggestion
  works fine as a standalone program in your local and CI environments.
```

File: docs/getting-started/comparison.md [hunk 3]
```diff
@@ -66,6 +66,47 @@ There are a number of differences between Dojo and Batect:
...
+- Earthly has a larger support community
+- Earthly is more focused on Docker, less abstracted from details than Batect
+- Caching has overlapping and dissimilar features.
+  Earthly is closer to Docker-style caching of layers
```
Comment (charleskorn) [inline docs/getting-started/comparison.md:98]:
```suggestion
- Earthly and Batect have overlapping but also different caching features.
  Earthly is closer to Docker-style caching of layers.
```

File: docs/getting-started/comparison.md [hunk 4]
```diff
@@ -66,6 +66,47 @@ There are a number of differences between Dojo and Batect:
...
+- Earthly configuration uses syntax like that of `Dockerfile` (called
+  `Earthfile`) with some syntax borrowed from `Makefile` (but without the
+  obsession for `<TAB>`s).
+  Batect uses YAML for configuration syntax
```
Comment (charleskorn) [inline docs/getting-started/comparison.md:102]:
```suggestion
  Batect uses YAML for configuration syntax.
```

File: docs/getting-started/comparison.md [hunk 5]
```diff
@@ -66,6 +66,47 @@ There are a number of differences between Dojo and Batect:
...
+- Earthly and Batect cache for you in different situations.
+  Earthly tends to be more automatic for caching
```
Comment (charleskorn) [inline docs/getting-started/comparison.md:104]:
```suggestion
  Earthly tends to be more automatic for caching.
```

File: docs/getting-started/comparison.md [hunk 6]
```diff
@@ -66,6 +66,47 @@ There are a number of differences between Dojo and Batect:
...
+  works fine as a standalone program in your local and CI environments
+- Earthly has a larger support community
+- Earthly is more focused on Docker, less abstracted from details than Batect
+- Caching has overlapping and dissimilar features.
+  Earthly is closer to Docker-style caching of layers
+- Earthly configuration uses syntax like that of `Dockerfile` (called
+  `Earthfile`) with some syntax borrowed from `Makefile` (but without the
+  obsession for `<TAB>`s).
+  Batect uses YAML for configuration syntax
+- Earthly and Batect cache for you in different situations.
```
Comment (charleskorn) [inline docs/getting-started/comparison.md:103]: I'd probably move this point up to near the earlier point about caching features.

File: docs/getting-started/comparison.md [hunk 7]
```diff
@@ -66,6 +66,47 @@ There are a number of differences between Dojo and Batect:
...
+  Earthly tends to be more automatic for caching
+- Earthly is very verbose, and does not have options to quiet the output.
+  (Try the `-o quiet` option in Batect)
```
Comment (charleskorn) [inline docs/getting-started/comparison.md:106]:
```suggestion
- Earthly is very verbose, and does not have options to quiet the output, whereas Batect has the `--output=quiet` option to show only minimal output.
```

File: docs/getting-started/comparison.md [hunk 8]
```diff
@@ -66,6 +66,47 @@ There are a number of differences between Dojo and Batect:
...
+- Configuration for Earthly can be more specific than Batect, but as a result
+  also be more flexible. Each project needs to evaluate this for themselves
```
Comment (charleskorn) [inline docs/getting-started/comparison.md:108]:
```suggestion
- Configuration for Earthly can be more specific than Batect, but as a result
  also be more flexible. Each project needs to evaluate this for themselves.
```

**Review by charleskorn — COMMENTED** [review]
Thanks for the PR @binkley. Overall, this looks good, I have just a couple of formatting nits.

**Comment by binkley** [comment]
Pushed on the PR. There are language improvements along suggestions. The key thing I note is a style choice: end bullet points with full stops; ensure they are proper sentences.

**Review by charleskorn — APPROVED** [review]
Looks good to me, thanks @binkley. Could you please fix the linting issues (run `./batect lintFix`)?

**Comment by binkley** [comment]
I'm not sure what's getting fixed?
```
🌷 ./batect fix-lint
Running fix-lint...
node-build-env: running yarn lintFix

yarn run v1.22.19
$ eslint --fix src *.js *.ts
/bin/sh: 1: eslint: not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

fix-lint finished with exit code 127 in 2.9s.
(Exit 127) ./batect fix-lint
```

**Comment by binkley** [comment]
Running `./batect fix-lint` showed no deltas.

**Comment by charleskorn** [comment]
Not all linting issues can be fixed automatically - you may need to manually fix them.

**Comment by binkley** [comment]
FYI -- the `setup` task does not install `eslint`

```
🌷 ./batect lint
Version 0.83.6 of Batect is now available (you have 0.83.5).
To upgrade to the latest version, run './batect --upgrade'.
For more information, visit https://github.com/batect/batect/releases/tag/0.83.6.

Running lint...
node-build-env: running yarn lint

yarn run v1.22.19
$ eslint src *.js *.ts && remark . --ext md,mdx --quiet --frail --rc-path .remarkrc.mjs && editorconfig-checker
/bin/sh: 1: eslint: not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

lint finished with exit code 127 in 2.1s.
(Exit 127) ./batect lint
```

**Comment by charleskorn** [comment]
I've just tried with a fresh checkout and `./batect setup` and then `./batect lint` works OK... could you try running `./batect --clean` and then `./batect setup` and `./batect lint`?

**Comment by binkley** [comment]
Perhaps I overthought this.
```
$ ./batect setup
$ ./batect lint
```
This works. I'll update my doc edits and push. Previously I was getting lint warnings about JS/TS files which didn't make sense.

File: docs/getting-started/comparison.md [hunk 9]
```diff
@@ -66,6 +66,63 @@ There are a number of differences between Dojo and Batect:
...
+- Earthly and Batect have overlapping and also different caching features.
+  Earthly is closer to Docker-style caching of layers.
+
+- Earthly and Batect cache for you in different situations.
+  Earthly tends to be more automatic for caching but also less under your
+  control without additional configuration.
+
+- Earthly configuration uses syntax like that of `Dockerfile` (called
+  `Earthfile`) with some syntax borrowed from `Makefile` (but without the
+  obsession for `<TAB>`s).
```
Comment (charleskorn) [inline docs/getting-started/comparison.md:115]:
```suggestion
- Earthly configuration uses syntax like that of `Dockerfile` (called
  `Earthfile`) with some syntax borrowed from `Makefile`.
```

**Comment by charleskorn** [comment]
> Why the rule on newlines between bullet items?

This is very much a personal preference thing - for me, it's easier to read longer lists with blank lines between them.

**Review by charleskorn — APPROVED** [review]
This looks good to me, thanks for submititng this @binkley!

**Pull request merged**